src/Analyzer.Utilities/CSharp/ITypeSymbolExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Analyzer.Utilities
+{
+    public static class ITypeSymbolExtensions
+    {
+        public static bool IsAssignableTo(this ITypeSymbol fromSymbol, ITypeSymbol toSymbol, Compilation compilation)
```

Comment:
However, this is not to push back on your current effort of adding C# and VB specific utilities. :+1: on that. We can probably move common syntax related helper functions from all C# and VB analyzer project to these assemblies.